### ISSUE TITLE
Way to make dock widget not floatable

### ISSUE THREAD

**Issue by dlinenbe** [KDAB/KDDockWidgets#524]
Is there a way to make a dock widget not floatable?  I would like to make a dock widget, for example, only able to be docked to the right side or the left side - but never top or bottom - and never be allowed to float.  Is this possible?

David Linenberg

**Comment by iamsergio** [comment]
I've improved the readme: https://github.com/KDAB/KDDockWidgets/blob/main/docs/book/src/dnd.md

those are the available functions

**Comment by dlinenbe** [comment]
The docs do not make it clear at all how to make a widget only be dockable on the right side of the main window, for example. Or be movable but not floatable - just like a toolbar. A small code snippet would be extremly gratifying, assuming this can be done. Thank you!

**Comment by iamsergio** [comment]
# Docking only to left/right

Example here: `qtwidgets_dockwidgets --hide-certain-docking-indicators`

# Not make it float

If you start a DnD then it floats. What you want is not possible in an elegant way.
Maybe could be achieved if you programmatically redock it. `dock->setFloating(false)` should put it back where it was

**Comment by iamsergio** [comment]
There's a `DragController::instance()->isDraggingChanged` signal could be the right moment to redock it

**Comment by dlinenbe** [comment]
I  have tried things like within the drag about to start function, querying if  the draggable is a single widget - then if so, saving that dockWidget in a variable - and when we enter the dragging end function - if that variable is set, we can set floating for that dockWidget to false. So that works for a single window.   But there are issues - for example, when we overlay widgets on top of each other - and  then move a tabbed group of widgets outside of the main window - then it becomes tricky enough that I have not figured out how to elegantly solve the problem.

**Comment by iamsergio** [comment]
I'll give it a try in the example

**Comment by iamsergio** [comment]
```
        KDDockWidgets::Config::self().setDragEndedFunc([] {
            if (s_dragged) {
                if (auto *tb = qobject_cast<KDDockWidgets::Core::TitleBar *>(s_dragged->asController())) {
                    if (tb->isFloating()) {
                        tb->onFloatClicked();
                    }
                } else {
                    qDebug() << "TODO: handle " << s_dragged->asController(); // probably tab bar case
                }
            }
        });

        KDDockWidgets::Config::self().setDragAboutToStartFunc([](Core::Draggable *draggable) {
            s_dragged = draggable;
            return true;
        });
```

Seems to work for the tabbed group case as well

**Comment by dlinenbe** [comment]
Yes that seems pretty good. It seems to not have the desired behaviour when one docks s one window over another (so tab bar with 
 2 windows). Then take one of the tabbed windows and move  it externally to to the main window - it allows the window to remain outside.

**Comment by iamsergio** [comment]
```cpp
        KDDockWidgets::Config::self().setDragEndedFunc([] {
            const auto floatingWindows = KDDockWidgets::DockRegistry::self()->floatingWindows();
            for (auto fw : floatingWindows) {
                if (!fw->beingDeleted())
                    fw->titleBar()->onFloatClicked();
            }
        });
```

This seems to work for all cases

**Comment by dlinenbe** [comment]
That seems to work great! Thanks.  My final issue is that for some reason, double click on the titlebar (Windows, QT 6.8.0) will make the window floating...
` auto &config = KDDockWidgets::Config::self();
  config.setFlags(config.flags() | KDDockWidgets::Config::Flag_TitleBarNoFloatButton);
  config.setFlags(config.flags() &~KDDockWidgets::Config::Flag_DoubleClickMaximizes);  \\ does not work for some reason`

**Comment by dlinenbe** [comment]
Thank you!  My last issue is that for some reason, I cannot disable the double clicking on the titlebar (Windows 11, QT 6.8.0). It creates a floating window, even with this code: 
`auto &config = KDDockWidgets::Config::self();
  config.setFlags(config.flags() & ~KDDockWidgets::Config::Flag_DoubleClickMaximizes);`

**Comment by iamsergio** [comment]
Seems there's no way, unless you override the TitleBar, and override `mouseDoubleClickEvent()`
there's an example of custom titlebar in `dockwidgets/MyViewFactory.cpp`, or you could use a global event filter and discard it if the target is a titlebar
